src/MWethDelegate.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.19;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {WETH9} from "@protocol/router/IWETH.sol";
+import {WethUnwrapper} from "@protocol/WethUnwrapper.sol";
+import {MErc20Delegate} from "@protocol/MErc20Delegate.sol";
+
+/**
+ * @title Moonwell's MWethDelegate Contract
+ * @notice MToken which wraps underlying ETH
+ * @author Moonwell
+ */
+contract MWethDelegate is MErc20Delegate {
+    using SafeERC20 for IERC20;
+
+    /// @notice the WETH unwrapper address
+    address constant public wethUnwrapper = 0xcd57F6D5162174D16b0215841B4414fF2A0441Ac;
```

Comment:
I guess you don't want to change the constructor of the MErc20Delegate here, thats why its hard coded?

Reply:
no, the reason it's hardcoded is because this contract is delegatecalled by the delegator contract, so if it wasn't constant, we would need to write to storage in the delegator to store this variable, or upgrade the delegator which would discard the old instance.

if we set it in the constructor, it would be either immutable or storage and the delegator wouldn't know the address when it delegate called in